[GRID 3.2] Add base theming syntax for the grid

Koerty opened the pull request:
**Changes**
**EDIT : removed "margin, tileMaxSize, selectedTileMaxSize" for now**
- The themes can now configure following elements : pos, size
- Change some default values for them

**Testing**
You can test by using [my custom test theme](https://github.com/Koerty/es-theme-test-gridview), it had a green border around the grid so you can better see what's going on (like on the screenshots of my [previous PR](https://github.com/RetroPie/EmulationStation/pull/396)).

Or using this modified version of @lilbud Material theme : [here](https://github.com/Koerty/es-theme-material)

Or by creating a new theme.xml from this : 
```xml
<theme>
    <formatVersion>4</formatVersion>
	<view name="grid">
		<imagegrid name="gamegrid">
			<pos>0.1 0.1</pos>
			<size>0.8 0.8</size>
<!-- Removed for now
			<margin>0.01 0.01</margin>
			<tileMaxSize>0.19 0.19</tileMaxSize>
			<selectedTileMaxSize>0.22 0.22</selectedTileMaxSize>
-->
		</imagegrid>
	</view>
</theme>
```

If a native english speaker could check the changes made to the file THEMES.md, it would be nice. I'm not so proud of this sentence for example : 
`
* `size` - type: NORMALIZED_PAIR.
    - The size of the grid. Take care the selected tile can go out of the grid size, so don't position the grid too close to another element or the screen border.
`

lilbud commented:
>also, what is correct term for the blank space between tiles ? Is it the grid "margin" or the grid "padding" ?

Last time I screwed with gridview, it was the margin.

hex007 commented:
@lilbud its called margin. 

The difference between padding and margin is subtle. Padding is inside the component while margin is outside the component. Sometimes components are clipped to padding. This allows non overlapping animations like the carousel system text.

Koerty commented:
Thank you for your input.

Koerty commented:
I think we should delay this PR and focus on the others.

The reason is the 2 PR related to metadata and some other smaller stuff is ready for review, and while you review that, it will give me more time to finish to (re)implement the missing base functionalities (tile background, default images for game, etc ...), thus giving me also more time to take a second, more mature, look to the new theming syntax proposition, and do more testing about it.

Maybe I won't change anything about the syntax, but I don't think I can guarantee I will not, that's why I think this is the best strategy.

jrassa commented:
At a minimum, we may want to go ahead and implement support for grid size and position, since I don't think those are likely to change.  With #404 merged, the metadata can be positioned, but you can move/resize the grid so the metadata is on top of the grid.

Koerty commented:
@jrassa Sure. I was thinking more about tileMaxSize and selectedTileMaxSize, which I may move to another theme element called "gridtile" (a bit like my 2nd syntax proposition did). I changed this PR to just add "size" and "pos" (which are already handled by GuiElement).

I rebased and solved merge conflicts. There is just one last it to change, sorry for asking this but can you check the sentence I talked about at the end of my first message ?

After that, it's ready to get merged.

jrassa commented:
Looks good.  Thanks.
